### ISSUE TITLE
conman and log_server: Idempotency issues

### ISSUE THREAD

**Issue by osmocl** [bluebanquise/bluebanquise#467]
Description
-------------

Hello guys, while running automatic tests using kitchen, we have found two idempotency issues in a management node.
Second running is always changed for the following two tasks:

log_server role fail in : `task 'Create /var/log/rsyslog directory'`
conman role fail in : `task 'Create /var/log/conman directory'`

Info
----
OS: RHEL8.2
Branch: master

Reproducer
--------------

Our playbook : 

```
# cat playbooks/managements.yml
---

- name: managements playbook
  hosts: "management1"
  roles:

    - role: set_hostname
      tags: set_hostname
    - role: nic
      tags: nic
    - role: repositories_server
      tags: repositories_server
    - role: repositories_client
      tags: repositories_client
    - role: bluebanquise
      tags: bluebanquise
    - role: hosts_file
      tags: hosts_file
    - role: ssh_master
      tags: ssh_master
    - role: dhcp_server
      tags: dhcp_server
    - role: dns_server
      tags: dns_server
    - role: dns_client
      tags: dns_client
    - role: time
      tags: time
      vars:
        time_profile: server
    - role: pxe_stack
      tags: pxe_stack
    - role: diskless
      tags: diskless
    - role: nfs_server
      tags: nfs_server
    - role: prometheus_server
      tags: prometheus_server
    - role: prometheus_client
      tags: prometheus_client
    - role: grafana_server
      tags: grafana_server
    - role: log_server
      tags: log_server
    - role: conman
      tags: conman
```

Command:

For conman role, first execution:

```
# ansible-playbook playbooks/managements.yml -t conman
# ll -Z /var/log/conman
total 8
-rw-------. 1 root root system_u:object_r:conman_log_t:s0 445 Sep 24 10:28 c001.log
-rw-------. 1 root root system_u:object_r:conman_log_t:s0 480 Sep 24 10:28 management1.log
```

Second execution, as you can see the log permissions has changed:

```
# ansible-playbook playbooks/managements.yml-t conman 
changed=1 > TASK [file █ Create /var/log/conman directory

# ll -Z /var/log/conman
total 8
-rwxr-x---. 1 root root system_u:object_r:conman_log_t:s0 445 Sep 24 10:28 c001.log
-rwxr-x---. 1 root root system_u:object_r:conman_log_t:s0 480 Sep 24 10:28 management1.log
```

For log_server role, first play:

```
# ansible-playbook playbooks/managements.yml-t log_server 
# ll /var/log/rsyslog/management1/
total 132
-rw-------. 1 root root   214 Nov 10 13:07 anacron.log
-rw-------. 1 root root   531 Nov 10 13:07 ansible-copy.log
-rw-------. 1 root root   579 Nov 10 13:07 ansible-dnf.log
-rw-------. 1 root root  1094 Nov 10 13:07 ansible-file.log
-rw-------. 1 root root  2705 Nov 10 13:07 ansible-firewalld.log
-rw-------. 1 root root  1507 Nov 10 13:07 ansible-lineinfile.log
-rw-------. 1 root root   328 Nov 10 13:07 ansible-seport.log
-rw-------. 1 root root   720 Nov 10 13:07 ansible-setup.log
-rw-------. 1 root root   385 Nov 10 13:07 ansible-stat.log
-rw-------. 1 root root   412 Nov 10 13:07 ansible-systemd.log
-rw-------. 1 root root   656 Nov 10 13:07 conmand.log
-rwxr-x---. 1 root root   234 Nov 10 13:07 dbus-daemon.log
-rw-------. 1 root root   213 Nov 10 13:07 firewalld.log
-rw-------. 1 root root 40710 Nov 10 13:07 karma.log
-rwxr-x---. 1 root root  3423 Nov 10 13:02 named.log
-rw-------. 1 root root   306 Nov 10 13:07 polkit-agent-helper-1.log
-rw-------. 1 root root   757 Nov 10 13:07 polkitd.log
-rw-------. 1 root root   471 Nov 10 13:07 rsyslogd.log
-rw-------. 1 root root   825 Nov 10 13:07 sshd.log
-rw-------. 1 root root  3038 Nov 10 13:07 sudo.log
-rw-------. 1 root root   307 Nov 10 13:07 su.log
-rw-------. 1 root root  1747 Nov 10 13:07 systemd.log
-rw-------. 1 root root   328 Nov 10 13:07 systemd-logind.log
-rw-------. 1 root root   121 Nov 10 13:07 systemd-udevd.log
```

second play:

```
# ansible-playbook playbooks/managements.yml-t log_server 
# ll /var/log/rsyslog/management1/
total 144
-rwxr-x---. 1 root root   214 Nov 10 13:07 anacron.log
-rwxr-x---. 1 root root   531 Nov 10 13:07 ansible-copy.log
-rwxr-x---. 1 root root  1158 Nov 10 13:08 ansible-dnf.log
-rwxr-x---. 1 root root  2739 Nov 10 13:08 ansible-file.log
-rwxr-x---. 1 root root  3246 Nov 10 13:08 ansible-firewalld.log
-rwxr-x---. 1 root root  3014 Nov 10 13:08 ansible-lineinfile.log
-rwxr-x---. 1 root root   656 Nov 10 13:08 ansible-seport.log
-rwxr-x---. 1 root root   864 Nov 10 13:08 ansible-setup.log
-rwxr-x---. 1 root root   770 Nov 10 13:08 ansible-stat.log
-rwxr-x---. 1 root root   617 Nov 10 13:08 ansible-systemd.log
-rwxr-x---. 1 root root   656 Nov 10 13:07 conmand.log
-rwxr-x---. 1 root root   234 Nov 10 13:07 dbus-daemon.log
-rwxr-x---. 1 root root   213 Nov 10 13:07 firewalld.log
-rwxr-x---. 1 root root 52772 Nov 10 13:08 karma.log
-rwxr-x---. 1 root root  3973 Nov 10 13:08 named.log
-rwxr-x---. 1 root root   306 Nov 10 13:07 polkit-agent-helper-1.log
-rwxr-x---. 1 root root   757 Nov 10 13:07 polkitd.log
-rwxr-x---. 1 root root   471 Nov 10 13:07 rsyslogd.log
-rwxr-x---. 1 root root   825 Nov 10 13:07 sshd.log
-rwxr-x---. 1 root root  3559 Nov 10 13:08 sudo.log
-rwxr-x---. 1 root root   307 Nov 10 13:07 su.log
-rwxr-x---. 1 root root  1747 Nov 10 13:07 systemd.log
-rwxr-x---. 1 root root   328 Nov 10 13:07 systemd-logind.log
-rwxr-x---. 1 root root   121 Nov 10 13:07 systemd-udevd.log
```

We are suspecting an error due to `recurse: yes `option in the following tasks:

```
# cat conman/tasks/main.yml

- name: file █ Create /var/log/conman directory
  file:
    path: /var/log/conman
    mode: 0750
    setype: conman_log_t
    state: directory
    recurse: yes
  when:
    - ansible_facts.os_family == "RedHat"
    - ansible_facts.selinux.status == "enabled"

# cat roles/core/log_server/tasks/main.yml

- name: Create /var/log/rsyslog directory
  file:
    path: /var/log/rsyslog
    mode: 0750
    setype: var_log_t
    state: directory
    recurse: yes
  when:
    - ansible_facts.os_family == "RedHat"
    - ansible_facts.selinux.status == "enabled"
```

Removing the `recurse: yes` option allow us to run properly idempotence in the whole playbook including `conman `and `log_server `roles.

Any thoughts ?

**Comment by btravouillon** [comment]
I guess the `recurse` is useful to ensure the SELinux context on the files is correct. This is necessary in a use case where a user would enable SELinux after the initial deployment. However, this collapses with the `mode`. A proper fix would be to ensure files' mode is 0400, not 0750.

**Comment by btravouillon** [comment]
I forgot about this bug report. 😯 
The `mode: 0750` was added in d70ad62 to comply with a [new rule](https://ansible-lint.readthedocs.io/en/latest/default_rules.html#file-permissions-unset-or-incorrect) introduced in ansible-lint recently. This was a bad move, my bad. In #496, I removed the `mode` and added the `# noqa 208` to skip the test for this specific task.
I will open a PR for log_server to apply the same change.
